frontend/src/views/NewStore/BusinessForm.js

```diff
@@ -1,115 +1,233 @@
-  return needle ? needle.long_name : '';
+  const needle = haystack.find(item =>
+    item.types.some(type => type === wantedType)
+  );
+  return needle ? needle.long_name : "";
 };
 
 const BusinessForm = ({ location, history }) => {
-  const state = !location.state ? 'init' :
-    location.state.useGoogle ? 'google' : 'manual';
+  const state = !location.state
+    ? "init"
+    : location.state.useGoogle
+    ? "google"
+    : "manual";
   const { token, user } = useUserContext();
 
   const [data, setData] = useState({
-    name: '',
-    mail: '',
-    website: '',
-    phone: '',
-    country: 'DE',
-    line1: '',
-    line2: '',
-    description: '',
-    mapsPlaceObject: {},
+    name: "",
+    mail: "",
+    website: "",
+    phone: "",
+    country: "DE",
+    line1: "",
+    line2: "",
+    latitude: 0,
+    longitude: 0,
+    description: "",
+    mapsPlaceObject: {}
   });
 
-  const changeHandler = (event) => {
-    let _data = {...data, mapsPlaceObject: false};
+  const changeHandler = event => {
+    let _data = { ...data, mapsPlaceObject: false };
 
-    if (event === false)
-      return setData(_data);
+    if (event === false) return setData(_data);
 
     if (event.target) {
       _data = { [event.target.name]: event.target.value };
     } else {
       _data = {
-        name: event.structured_formatting && event.structured_formatting.main_text,
+        name:
+          event.structured_formatting && event.structured_formatting.main_text,
         website: event.website,
         phone: event.formatted_phone_number,
-        line1: getItemFromAddress('route', event.address_components) + ' ' +
-          getItemFromAddress('street_number', event.address_components),
-        line2: getItemFromAddress('postal_code', event.address_components) + ' ' +
-          getItemFromAddress('locality', event.address_components),
-        mapsPlaceObject: event,
+        line1:
+          getItemFromAddress("route", event.address_components) +
+          " " +
+          getItemFromAddress("street_number", event.address_components),
+        line2:
+          getItemFromAddress("postal_code", event.address_components) +
+          " " +
+          getItemFromAddress("locality", event.address_components),
+        mapsPlaceObject: event
       };
     }
 
     setData({ ...data, ..._data });
   };
 
-  const submitHandler = async (e) => {
+  const submitHandler = async e => {
     e.preventDefault();
-    const response = await API.partners.post({
-      ...data, address: { ...data }, users: [user.id]
-    },{headers: {'Authorization': `Token ${token}`}});
+    let searchString = data.line1 + " , " + data.line2;
+    const getLocation = await axios
+      .get(
+        `https://eu1.locationiq.com/v1/search.php?key=${process.env.REACT_APP_KEY}&q=${searchString}&format=json`
+      )
+      .then(response => {
+        setData({
+          ...data,
+          latitude: response.data[0].lat,
+          longitude: response.data[0].lon
```

Comment:
You'll have to trim the digits: 
![image](https://user-images.githubusercontent.com/16354712/80250177-99ec9080-8673-11ea-9859-ccb8fe0c591b.png)